java/dataset/src/main/cpp/jni_wrapper.cc

```diff
@@ -458,8 +467,8 @@ JNIEXPORT void JNICALL Java_org_apache_arrow_dataset_jni_JniWrapper_closeDataset
  * Signature: (J[Ljava/lang/String;JJ)J
  */
 JNIEXPORT jlong JNICALL Java_org_apache_arrow_dataset_jni_JniWrapper_createScanner(
-    JNIEnv* env, jobject, jlong dataset_id, jobjectArray columns, jlong batch_size,
-    jlong memory_pool_id) {
+    JNIEnv* env, jobject, jlong dataset_id, jobjectArray columns,
+    jobject substrait_extended_expression, jlong batch_size, jlong memory_pool_id) {
```

Comment:
Instead of passing `substrait_extended_expression`, can we pass in `filter` and `projection` parameters?

Reply:
It is possible to do that. It may be necessary to invoke Cpp twice to get bounded expressions, once for each filter and once for each projection. What do you think?

Reply:
I think that should be fine. At this point we have already crossed the JNI boundary so I don't anticipate much performance impact in making the call twice. Ideally, we should keep the implementation flexible enough such that somebody can easily add support for Acero compute expressions as well. e.g. from the user perspective a `filter` can be either a compute expression or a substrait binary blob that gets parsed into a compute expression (wrapped in named/bounded expression objects). We don't need to add that additional functionality in this PR, though.

Reply:
Changed